proposals/0026-hlsl-long-vector-type.md

```diff
@@ -136,6 +136,58 @@ myBuffer.Store(elementIndex, val);
 Long vectors support the existing vector subscript operators `[]` to access the scalar element values.
 They do not support any swizzle operations.
 
+#### Converting dimensions of long vectors
+
+HLSL vectors support extraction and recomposition of new vectors via swizzling, however
+long vectors don't support swizzling instead two new operations are made available on
+all vectors.
+
+**`slice<Offset, Count, ComponentType, InSize>(vector<ComponentType, InSize> in)`**
+
+The `slice` function returns a new `vector` containing elements starting at `Offset`
+containing `Count` number of components of type `ComponentType`. `ComponentType` and
+`InSize` are deduced by the compiler.
+
+```hlsl
+vector<int, 6> src = {0, 1, 2, 3, 4, 5};
+vector<int, 3> sub = slice<2, 3>(src) // sub = {2, 3, 4}
+```
+
+`slice` also has a convenience version that assumes Offset is 0.
+
+```hlsl
+vector<int, 6> src = {0, 1, 2, 3, 4, 5};
+vector<int, 4> sub = slice<4>(src) // sub = {0, 1, 2, 3}
+```
+
+A compiler error will be raised if the slice operation would result in out of bounds access.
+
+**`shuffle<Idx1, Idx2, ..., ComponentType, InSize>(vector<ComponentType, InSize> in)`**
+
+*`shuffle` is only available in HLSL 202y*
```

Comment:
We should remove `shuffle` from this proposal, and file an issue on it to re-evaluate with HLSL 202y. I don't want to lump this in with long vectors since we're supporting long vectors in DXC, and we are not supporting HLSL 202y in DXC.

Reply:
Done! https://github.com/microsoft/hlsl-specs/issues/599